contracts/RelayHub.sol

```diff
@@ -114,24 +117,30 @@ contract RelayHub is IRelayHub {
         emit Withdrawn(account, dest, amount);
     }
 
-    function verifyGasLimits(
-        uint256 paymasterMaxAcceptanceBudget,
+    function calldataGasCost(uint256 length) public override view returns (uint256) {
+        return dataGasCostPerByte.mul(length);
+}
+
+    function verifyGasAndDataLimits(
+        uint256 maxRelayExposure,
         GsnTypes.RelayRequest calldata relayRequest,
         uint256 initialGas
     )
     private
     view
-    returns (IPaymaster.GasLimits memory gasLimits, uint256 maxPossibleGas) {
-        gasLimits =
-            IPaymaster(relayRequest.relayData.paymaster).getGasLimits{gas:50000}();
-
-        require(paymasterMaxAcceptanceBudget >= gasLimits.acceptanceBudget, "unexpected high acceptanceBudget");
+    returns (IPaymaster.GasAndDataLimits memory gasAndDataLimits, uint256 maxPossibleGas) {
+        gasAndDataLimits =
+            IPaymaster(relayRequest.relayData.paymaster).getGasAndDataLimits{gas:50000}();
+        require(msg.data.length <= gasAndDataLimits.calldataSizeLimit, "msg.data exceeded limit" );
+        uint256 dataGasCost = calldataGasCost(msg.data.length);
+        require(maxRelayExposure >= gasAndDataLimits.acceptanceBudget.add(dataGasCost), "unexpected high maxRelayExposure");
```

Comment:
the unexpected value is the acceptanceBudget+dataCost, not the maxRelayExposure

Reply:
fixed